geth/mailservice/mailservice_api.go

```diff
@@ -64,6 +82,90 @@ func setMessagesRequestDefaults(r *MessagesRequest) {
 	}
 }
 
+// waitPeerAdded tries to connect with a peer and waits till connection will be established.
+func (api *PublicAPI) waitPeerAdded(peer *discover.Node, timeout time.Duration) error {
```

Comment:
Since this function isn't only waiting, but also adds a peer, I'd call it just `addPeer`. The comment is enough for specifying that it waits.